General Updates 12

onein528 opened the pull request:
### Before you start
- [x] Have you checked if a similar PR has already been requested? 
- [x] Have you built and ran the app?

### Related/Fixed issues

- Closes #182

### Description

Official General Update round 12

### Motivation and Context

#### Todo

- [ ] Add click event to activities button in activities section in home page.
- [ ] Add click event to feed items in activities page.
- [ ] Handle OAuth Restriction exception properly.
- [ ] Completely update and improve indivisual issue/PR page.
- [ ] Add repository Insights page
- [ ] Add organization Peoples page
- [ ] Add organization Projects page
- [ ] Add filtering feature (reposityries, users, notifications, ...)
- [ ] Update indivisual Discussion page
- [ ] Update indivisual Project page
- [ ] Add repository Actions page
- [ ] Add search page

#### Bugs known

_None_

### Assets (if appropriate):

Redesigned sign in page

![image](https://user-images.githubusercontent.com/62196528/175802853-9eda2f5d-90d6-456e-bb01-581b7acce112.png)

onein528 commented:
![image](https://camo.githubusercontent.com/c4d8e324d1cf1546388188e5610fa40c2f0d2b2a56b6c21cc99a7bbafd5c11f3/68747470733a2f2f6c75616e646572736f6e6e2e636f6d2f696d616765732f6d796674702f73637265656e73686f74732f6d796674702d6c696768742e706e67)

We can like this? [MyFTP](https://github.com/luandersonn/MyFTP) using muxc:TwoPaneView
Code https://github.com/luandersonn/MyFTP/blob/main/MyFTP/Views/LoginViewPage.xaml#L52-L259

DeveloperWOW64 commented:
> We can like this? [MyFTP](https://github.com/luandersonn/MyFTP) using muxc:TwoPaneView Code https://github.com/luandersonn/MyFTP/blob/main/MyFTP/Views/LoginViewPage.xaml#L52-L259

@onein528 for login? if so, yes! I love the look

onein528 commented:
@DeveloperWOW64 Yes. problem is we have no image fitting this pane mode.

DeveloperWOW64 commented:
close accidental?

onein528 commented:
yep.

DeveloperWOW64 commented:
oops

DeveloperWOW64 commented:
> @DeveloperWOW64 Yes. problem is we have no image fitting this pane mode.

I can generate one later

DeveloperWOW64 commented:
@onein528 Done! I can't code now but heres the image:
![FluentHub Hero v2](https://user-images.githubusercontent.com/71598437/174442449-69d5f89a-b8dd-41c3-a19c-eefbe10303e5.png)
It's exactly the same size as the one in MyFTP.

onein528 commented:
Thanks!!

DeveloperWOW64 commented:
ill dev now

DeveloperWOW64 commented:
> Shall we use StyledOction for primer/octions. octions does not expect to use as font(abandoned long time ago). And so we should have as svg and handle those svg in StyledOction user control.
> 
> StuledOction is one of the github's design.

It will take time, but sure. @onein528, you do realise it's called _Octicons_ and not _Octions_

onein528 commented:
wow 'c' missed. haha thank you ;)

onein528 commented:
@DeveloperWOW64 @BobbyESP @luandersonn  

Do you guys have any ideas other than to merge this?

DeveloperWOW64 commented:
> Do you guys have any ideas other than to merge this?

@onein528 what's changed?

and what about these:
![image](https://user-images.githubusercontent.com/71598437/175763483-d1acfe8b-aa7f-4969-b2a8-64f9e4c3bdc5.png)

onein528 commented:
@DeveloperWOW64 Yes. I updated No.4 to some extent.

Other that those goals. ideas, or something.

DeveloperWOW64 commented:
> @DeveloperWOW64 Yes. I updated No.4 to some extent.
> 
> Other that those goals. ideas, or something.

okay. sure then, you can merge. ill review for you

onein528 commented:
No. I dont want to merge. if you have any ideas of something, you or I can add that feature.

DeveloperWOW64 commented:
> No. I dont want to merge. if you have any ideas of something, you or I can add that feature.

hmm.

DeveloperWOW64 commented:
Add click event to activities button in activities section in home page.

DeveloperWOW64 commented:
> Add click event to activities button in activities section in home page.

that should be easy, should it not?

DeveloperWOW64 commented:
Also i wish we could have this:
![image](https://camo.githubusercontent.com/c4d8e324d1cf1546388188e5610fa40c2f0d2b2a56b6c21cc99a7bbafd5c11f3/68747470733a2f2f6c75616e646572736f6e6e2e636f6d2f696d616765732f6d796674702f73637265656e73686f74732f6d796674702d6c696768742e706e67)
but it relies on certain WPF controls and FH is UWP

onein528 commented:
oh, right. do you have an thought about other UIs?

DeveloperWOW64 commented:
> oh, right. do you have an thought about other UIs?

MS Get started

onein528 commented:
Im sorry. not meant to. I want.. ah

Do you have any new UI concepts or ideas other than the intro page like repo page, issue page?

DeveloperWOW64 commented:
> Im sorry. not meant to. I want.. ah
> 
> Do you have any new UI concepts or ideas other than the intro page like repo page, issue page?

no actually

DeveloperWOW64 commented:
I saw this concept tho - it could be used in Insights
![image](https://user-images.githubusercontent.com/71598437/175764857-4d38a4ef-d88f-44fa-ba2a-a942e8861c4c.png)

onein528 commented:
wow it's very cool!! thanks!

DeveloperWOW64 commented:
I'm terrible at backend so I could do some UI if you want

DeveloperWOW64 commented:
@onein528 I'll dev in a bit - sry i've got other things to do

DeveloperWOW64 commented:
@onein528 what's the difference between using Octokit.GraphQL.NET and Octokit.Net?

onein528 commented:
Response speed. The second one allows me to get all data despite it is large amount. But first one allows me to get data specifically. For example, when I want to get specific repository owner only, it allows me.

DeveloperWOW64 commented:
> Response speed. The second one allows me to get all data despite it is large amount. But first one allows me to get data specifically. For example, when I want to get specific repository owner only, it allows me.

ah okay

DeveloperWOW64 commented:
Also, I'm not so sure Octokit.NET and Octokit.GraphQL.NET are abandonded - they both have recent commits

DeveloperWOW64 commented:
I feel i should dev as I have time but I do not know what to work on.

DeveloperWOW64 commented:
@onein528 Why don't we use strings?

onein528 commented:
due to massive UI updates.

DeveloperWOW64 commented:
> due to massive UI updates.

how are they related?

onein528 commented:
We had localization resources until GU2 was committed, but since GU3, the UI has noticeably changed a lot and the Uid has become more difficult to manage. We removed the resource because we often had to delete the resource and create a new one soon after we had it translated to Japanese, Spanish, or Chinese.

DeveloperWOW64 commented:
I think I may have found a way to code the MyFTP login screen.

DeveloperWOW64 commented:
> I think I may have found a way to code the MyFTP login screen.

nope. It works but it is very ugly and not like what I want it to be

DeveloperWOW64 commented:
@onein528 Are you deving?

onein528 commented:
Finally

![image](https://user-images.githubusercontent.com/62196528/175801669-30536a5b-2f02-4c1e-861d-3a00093ad952.png)

onein528 commented:
> @onein528 Done! I can't code now but heres the image: ![FluentHub Hero v2](https://user-images.githubusercontent.com/71598437/174442449-69d5f89a-b8dd-41c3-a19c-eefbe10303e5.png) It's exactly the same size as the one in MyFTP.

Actually, I dont think this hero image is good. kind of "image" is good like MyFTP using.

DeveloperWOW64 commented:
@onein528 why?

onein528 commented:
ah, window in window image is not good I think.

DeveloperWOW64 commented:
> ah, window in window image is not good I think.

why?

onein528 commented:
Nice hero. but little fluent

![image](https://user-images.githubusercontent.com/62196528/175801942-95a0a919-b390-4997-bbd9-83b5468eb7cc.png)

I dont know what is better. but MyFTP like image is good I think:

![vector1](https://user-images.githubusercontent.com/62196528/175801907-262216b1-b4f1-4a2a-a1b3-7f7de1b6c980.jpg)

DeveloperWOW64 commented:
Love the second one

onein528 commented:
Really. But i dont know it is legal to use.

DeveloperWOW64 commented:
> Really. But i dont know it is legal to use.

Where'd you get it?

DeveloperWOW64 commented:
If it's from shutterstock, 100% attribute. I don't want an app with an attribute.
My wallpaper library has over 75 wallpapers though.. Any wallpaper related to Windows would not need attribution as the user would be running the OS. Your profile pic, e.g.
![image](https://user-images.githubusercontent.com/71598437/175802184-219e811f-faa2-4ec9-8f6e-65b7768a0cea.png)

DeveloperWOW64 commented:
@onein528 What do you think?

onein528 commented:
Pick one up please.

(What's the "attribute"?)

DeveloperWOW64 commented:
> Pick one up please.
> 
> (What's the "attribute"?)

Attribution is when you need to say that you have obtained an asset (this case: image) from a place, rather than just use it.

Here is one of them:
![Surface_Laptop_3_07](https://user-images.githubusercontent.com/71598437/175802524-e59d39c8-b24a-49cb-bc36-ff03c4e1ee11.jpg)

onein528 commented:
you like this?

> ![](https://user-images.githubusercontent.com/71598437/175802524-e59d39c8-b24a-49cb-bc36-ff03c4e1ee11.jpg)

DeveloperWOW64 commented:
Yes. This is one of the Microsoft wallpapers, designed by the same team as WinUI i think, so it should fit well.

onein528 commented:
Do you think?

![image](https://user-images.githubusercontent.com/62196528/175802853-9eda2f5d-90d6-456e-bb01-581b7acce112.png)
![image](https://user-images.githubusercontent.com/62196528/175802859-fec4e445-5016-4f6d-a7ba-f37b59bf047b.png)

DeveloperWOW64 commented:
WOAH That looks fantastic!!

DeveloperWOW64 commented:
@onein528 Can you push?

onein528 commented:
Yep. a sec.

DeveloperWOW64 commented:
@onein528 hmm..
![image](https://user-images.githubusercontent.com/71598437/175805738-3477070d-d3d9-4ff7-9762-fe81951a07f6.png)

DeveloperWOW64 commented:
ありがとう！

DeveloperWOW64 commented:
@onein528 What do you want to complete before the merge of this PR? These?
![image](https://user-images.githubusercontent.com/71598437/175763483-d1acfe8b-aa7f-4969-b2a8-64f9e4c3bdc5.png)

onein528 commented:
first 3.

onein528 commented:
@DeveloperWOW64 Do you or others know something good app or web to create todo, milestone, or loadmap?

DeveloperWOW64 commented:
> @DeveloperWOW64 Do you or others know something good app or web to create todo, milestone, or loadmap?

@onein528 GitHub Projects is actually best because it is integrated so well. Projects also allows for 'iterations' - lots of PRs and issues linked to one feature/bug.

onein528 commented:
Ok. I will take a look at docs written how to use that.

DeveloperWOW64 commented:
> Ok. I will take a look at docs written how to use that.

you want to use it for FH? I can set it up.

onein528 commented:
yes. I have lots of To-do and its milestone, loadmap

onein528 commented:
v0.0 - v0.5 Done querying feature implementing(read)
v0.5 - v0.9 Done mutation feature implementing(edit)
v0.9 - v1.0 Prepare to publish stable version

DeveloperWOW64 commented:
Ok. I'll create these milestones - I might have to delete the current project tho

DeveloperWOW64 commented:
@onein528 This any good?
https://github.com/orgs/FluentHub/projects/4

DeveloperWOW64 commented:
@onein528 I tried to revert these because of the way GH Projects works. Once we're done with Query the Current channel will become mutation, even though it's called query.
![image](https://user-images.githubusercontent.com/71598437/175810914-21de965f-64ef-43a9-93ed-50b33062bee8.png)

onein528 commented:
Oh sorry. please revert it. I've changed.

DeveloperWOW64 commented:
done

DeveloperWOW64 commented:
@onein528 What do you mean by mutation?

onein528 commented:
refer to https://docs.github.com/en/graphql/reference/mutations

DeveloperWOW64 commented:
> refer to https://docs.github.com/en/graphql/reference/mutations

So the user can actually do stuff other than browse GH?

onein528 commented:
yep. edit user info, submit issue, add comment...

DeveloperWOW64 commented:
Cool! if you have any more features for planning, tell me.

onein528 commented:
WDYM??

DeveloperWOW64 commented:
> WDYM??

e.g. you want to add new feature that isn't one of the big 3.

onein528 commented:
hahah hhaaa.....

![image](https://user-images.githubusercontent.com/62196528/175811308-5ed080d7-49a0-4dc5-b844-ddea740b4d3e.png)

DeveloperWOW64 commented:
> hahah hhaaa.....

bruh

onein528 commented:
No, I will pick up from those.

onein528 commented:
1,2,3,4,5,6 Actually.

From those list you can also open and merge a PR.

DeveloperWOW64 commented:
> From those list you can also open and merge a PR.
> 
> 1,2,3,5,6,7 Actually.

WDYM?

onein528 commented:
You can commit more than one from those.

![image](https://user-images.githubusercontent.com/62196528/175811308-5ed080d7-49a0-4dc5-b844-ddea740b4d3e.png)

I want to commit No 1,2,3,5,6,7 into this.

DeveloperWOW64 commented:
> I want to commit No 1,2,3,5,6,7 into this.

You deving now?

onein528 commented:
yep. adding insights page.

DeveloperWOW64 commented:
> yep. adding insights page.

With the design I said or a different one?

onein528 commented:
but will get slow down another two weeks. please keep in mind.

DeveloperWOW64 commented:
also, just to be clear, are you starting on mutation/query in this PR?

onein528 commented:
> > yep. adding insights page.
> 
> With the design I said or a different one?

I dont know how it goes, but almost follow the one what you said. I really hate current GitHub Insights UI

onein528 commented:
> also, just to be clear, are you starting on mutation/query in this PR?

No. delete that.

DeveloperWOW64 commented:
> > also, just to be clear, are you starting on mutation/query in this PR?
> 
> No. delete that.

delete what?

onein528 commented:
from todo list

DeveloperWOW64 commented:
I'm working on the GH project - adding stuff, etc.

onein528 commented:
UI doesn't work properly despite I added UIs

onein528 commented:
Do you know why?

DeveloperWOW64 commented:
> Do you know why?

sry just building..

DeveloperWOW64 commented:
@onein528 hmmm (again)
![image](https://user-images.githubusercontent.com/71598437/175816154-710026ba-a6c1-4ee5-bc3f-f3efb6d1c566.png)

onein528 commented:
No way. can you clean solution and rebuild?

DeveloperWOW64 commented:
> No way. can you clean solution and rebuild?

nope. still doesn't work.

onein528 commented:
Can you debug?

DeveloperWOW64 commented:
> Can you debug?

ofc thats what Im doing. debugging is just reporting where the error is when you get it, instead of just crashing

onein528 commented:
so can you track why token is empty? I cannot track because this exception does not occur on my end.

DeveloperWOW64 commented:
> so can you track why token is empty? I cannot track because this exception does not occur on my end.

I know some, but I'm very bad at C#. Maybe it's because my XML document has wrong format.
Is this the same format as yours?
![image](https://user-images.githubusercontent.com/71598437/175816889-3daf353c-bf30-4e37-af6d-674dc19ab79d.png)

DeveloperWOW64 commented:
@onein528 Fixed it. Just some weird error. I reinstalled FH, which did the trick.

onein528 commented:
FluentHub is sometimes strange.

DeveloperWOW64 commented:
May have fixed the UI broken issue with [`d13045a`](https://github.com/FluentHub/FluentHub/pull/180/commits/d13045a0cf6150e2533a565d1f9a9f0f7a85f50a)

DeveloperWOW64 commented:
@onein528 now we need UI. Maybe we could use GridView?

onein528 commented:
> now we need UI. Maybe we could use GridView?

Talk about where?

And did you fixed insights page?

DeveloperWOW64 commented:
> > now we need UI. Maybe we could use GridView?
> 
> Talk about where?
> 
> And did you fixed insights page?

I think I fixed insights page - wdym by 'talk about where?'

onein528 commented:
> now we need UI.

What do you mean? talking about insighs page UI?

> I think I fixed insights page

Can you give me a screenshot of that?

DeveloperWOW64 commented:
> Can you give me a screenshot of that page?

@onein528 
![image](https://user-images.githubusercontent.com/71598437/175819616-1cb4cd3b-c6f7-424a-8fd7-3628f7be61e9.png)

onein528 commented:
On mine, literally broken.

![image](https://user-images.githubusercontent.com/62196528/175819689-ba1a1a6d-312e-446a-b2c3-637d7379c5e2.png)

DeveloperWOW64 commented:
> On mine, literally broken.

Pull from origin @onein528

onein528 commented:
Wow nice.

onein528 commented:
![image](https://user-images.githubusercontent.com/62196528/175821089-f3785217-ee75-4cf3-be85-e1033a8a31df.png)

DeveloperWOW64 commented:
I'll be able to dev in a bit

onein528 commented:
Going to sleep. bye.

DeveloperWOW64 commented:
> Going to sleep. bye.

さよなら ともゆき。

DeveloperWOW64 commented:
/azp run

DeveloperWOW64 commented:
Definitely broke fluenthub..
![image](https://user-images.githubusercontent.com/71598437/176609243-90ca6d91-e3f9-4ef9-b65a-44bdf4ec94ff.png)

onein528 commented:
I will create sign out methods. Please revert “reset” arg implemented commit. Cannot do like that.

onein528 commented:
```cs
Frame rootFrame = (Frame)Window.Current.Content;
rootFrame.Navigate(typeof(SignIn.IntroPage));
```

This code does not work?

DeveloperWOW64 commented:
> ```cs
> Frame rootFrame = (Frame)Window.Current.Content;
> rootFrame.Navigate(typeof(SignIn.IntroPage));
> ```
> 
> This code does not work?

no.

onein528 commented:
@luandersonn Do you know what kinda sliding animation is? In the situation where you want to change the content from “1” to “2”. Slide top to bottom.

onein528 commented:
2 comes from above and 1 go down

DeveloperWOW64 commented:
> @luandersonn Do you know what kinda sliding animation is? In the situation where you want to change the content from “1” to “2”. Slide top to bottom.

@onein528 winui3gallery://item/PageTransition or if you don't have winui 3 gallery, https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.Media.Animation.NavigationThemeTransition

onein528 commented:
Not page transition animation.

DeveloperWOW64 commented:
> Not page transition animation.

oh. as in content? It must be one of these:
![image](https://user-images.githubusercontent.com/71598437/176988852-e1e467ab-59ed-4011-af12-d6f7ddb9130c.png)

onein528 commented:
@DeveloperWOW64 Are you really not good at C#?

ioapiset reviewed (approved):
LGTM.

onein528 commented:
Actions at work.

DeveloperWOW64 commented:
> @DeveloperWOW64 Are you really not good at C#?

Not amazing, not terrible..

DeveloperWOW64 commented:
@onein528 Why the review?

DeveloperWOW64 commented:
o no maybe i broke it with [`c5b7fc5`](https://github.com/FluentHub/FluentHub/pull/180/commits/c5b7fc53cc05150496dee8def1d604fca9212785)

DeveloperWOW64 commented on src/FluentHub/FluentHub.csproj:
```diff
@@ -17,9 +17,18 @@
+    <AppxBundlePlatforms>x86|x64|arm</AppxBundlePlatforms>
+    <AppInstallerUri>https://api.codrex.dev/_get/_file/%24context=fluenthub/</AppInstallerUri>
+    <HoursBetweenUpdateChecks>0</HoursBetweenUpdateChecks>
   </PropertyGroup>
```
why? @onein528  they are required to generate the debug apppackages

onein528 replied:
then why did you allowed including pfx files?

DeveloperWOW64 replied:
> then why did you allowed including pfx files?

so it would build. it fails without.

DeveloperWOW64 commented:
> Changes summary:
> 
> ```
> + Added FUNDING.yml
> + Updated README.md
> + Added some schemas of Octokit.GraphQL
> + Added some url handling
> + Updated intro page
> + Added hero image
> + Improved some pages' code behind and those viewmodels
> + Added url handing to each page
> + Added some repo's insights page
> + Improved indivisual issue/PR page
> + Updated issue/PR's timeline
> ```
> 
> @DeveloperWOW64 Why did you add .pfx file?

certificate for apppackage debug. it will fail without it.

onein528 commented:
I see.

onein528 commented:
`No certificate found with the supplied thumbprint: 64B7A4274C47A261D02A8672BA06EEBD7926E47E`

Can you fix build errors?

DeveloperWOW64 commented:
> `No certificate found with the supplied thumbprint: 64B7A4274C47A261D02A8672BA06EEBD7926E47E`
> 
> Can you fix build errors?

should work with pfx file.

onein528 commented:
See latest pipeline result https://github.com/FluentHub/FluentHub/pull/180/checks?check_run_id=7200590807

onein528 commented:
on your end, it will success?

DeveloperWOW64 commented:
/azp run

DeveloperWOW64 commented:
> on your end, it will success?

yes

onein528 commented:
failed with same error.

DeveloperWOW64 commented:
@onein528 I will try on my machine.

onein528 commented:
And can you fix my grammer:?

```
+ Added FUNDING.yml
+ Updated README.md
+ Added some schemas of Octokit.GraphQL
+ Added some url handling
+ Updated intro page
+ Added hero image
+ Improved some pages' code behind and those viewmodels
+ Added url handing to each page
+ Added some repo's insights page
+ Improved indivisual issue/PR page
+ Updated issue/PR's timeline
```

DeveloperWOW64 commented:
Fixed:

```
+ Added FUNDING.yml
+ Updated README.md
+ Added schemas for Octokit.GraphQL
+ Added better support for URL handling
+ Updated Out of Box Experience
+ Added hero image (OOBE)
+ Improved some pages' code-behind and their ViewModels
+ Added Insights page
+ Improved individual issue/PR pages
+ Updated timeline of events in issues/PRs
```
@onein528

DeveloperWOW64 commented:
> failed with same error.

Weird. @onein528 My one works.
![image](https://user-images.githubusercontent.com/71598437/177381331-aa25f602-8314-4e50-9ea6-69b2fa0f5e49.png)

onein528 commented:
Can you revert c5b7fc53cc05150496dee8def1d604fca9212785 and 1360e59b08146e6804907e57a3f8b80e3d279f87 ?

DeveloperWOW64 commented:
> Can you revert [c5b7fc5](https://github.com/FluentHub/FluentHub/commit/c5b7fc53cc05150496dee8def1d604fca9212785) and [1360e59](https://github.com/FluentHub/FluentHub/commit/1360e59b08146e6804907e57a3f8b80e3d279f87) ?

sure.

onein528 commented:
When generate appx packages, always packag manager create those capabilities and pfx so i think dont need.

DeveloperWOW64 commented:
@onein528 Done with https://github.com/FluentHub/FluentHub/pull/180/commits/9b7b7adc813b92395f2165a64171779204934c17

ioapiset reviewed (approved):
LGTM.
